ambry-account/src/main/java/com/github/ambry/account/AccountServiceMetrics.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.account;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.MetricRegistry;
+
+
+/**
+ * {@link HelixAccountService} specific metrics tracking.
+ * <p/>
+ * Exports metrics that are triggered by the {@link HelixAccountService} to the provided {@link MetricRegistry}.
+ */
+public class AccountServiceMetrics {
+  // Histogram
+  public final Histogram startupTimeInMs;
+  public final Histogram accountUpdateTimeInMs;
+
+  // Counter
+  public final Counter unRecognizedMessageErrorCount;
+  public final Counter processFullAccountMessageErrorCount;
+  public final Counter nullZNRecordCount;
+  public final Counter nullAccountMapInZNRecordCount;
+  public final Counter buildAccountInfoMapFromRemoteRecordErrorCount;
+  public final Counter conflictWithRemoteRecordErrorCount;
+  public final Counter putAccountInAccountMapErrorCount;
+  public final Counter publishMessageErrorCount;
+  public final Counter updateAccountErrorCount;
+  public final Counter unExpectedNullZNRecordErrorCount;
+  public final Counter unExpectedNullAccountMapInZNRecordErrorCount;
+  public final Counter caseDConflictErrorCount;
+  public final Counter caseEConflictErrorCount;
+  public final Counter invalidRecordErrorCount;
+  public final Counter nullIdKeyErrorCount;
+  public final Counter unMatchedIdKeyAndAccountErrorCount;
+  public final Counter conflictInAccountInfoMapErrorCount;
```

Comment:
I'm not sure if we should have a counter for every type of error - looking at all the error counts might be almost impossible. Perhaps its better to have more general counters and descriptive log messages. We would then use the counters for monitoring and the log messages for the actual details.